subunit.go

```diff
@@ -127,10 +132,12 @@ func (e *Event) write(writer io.Writer) error {
 	var bTemp bytes.Buffer
 	bTemp.WriteByte(signature)
 	bTemp.Write(<-flagsChan)
-	bTemp.Write(<-timestampChan)
-	bTemp.Write(<-idChan)
-	bTemp.Write(<-mimeChan)
-	bTemp.Write(<-fileContentChan)
+	for _, part := range []packetPart{<-timestampChan, <-idChan, <-mimeChan, <-fileContentChan}{
```

Comment:
@fgimenez I have a question. Do you know if this blocks until the last element of the slice is created, or it starts consuming them (and potentially blocking) one by one?